### DISCUSSION TITLE
Is canvas of toga not available for android?

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1452] [Q&A]
Use canvas in the android app, but it raises "[Android] Not implemented: Canvas.new_path()". Is canvas of toga not available for android?

The code is as follows:

import toga
from toga.style import Pack

class HelloWorld(toga.App):
    def startup(self):
        self.main_window = toga.MainWindow(title=self.name)
        self.canvas = toga.Canvas(style=Pack(flex=1))
        box = toga.Box(children=[self.canvas], style=Pack(background_color="black"))

        with self.canvas.fill() as fill:
            fill.arc(50, 50, 15)
        self.main_window.content = box
        self.main_window.show()

def main():
    return HelloWorld()

The error as follows:

--------- beginning of system
06-20 22:34:14.171 17094 17094 D VRI[MainActivity]: mPopUpViewOffsets: offset=(0.0, 0.0), scale=(1.0, 1.0)
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: QUALCOMM build                   : eaa701cd06, I183f6321f0
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Build Date                       : 10/27/22
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.35.01.12
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Local Branch                     :
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Remote Branch                    :
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Remote Branch                    :
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Reconstruct Branch               :
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Build Config                     : S P 10.0.7 AArch64
06-20 22:34:14.171 17094 17117 I AdrenoGLES-0: Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
06-20 22:34:14.174 17094 17117 I AdrenoGLES-0: PFP: 0x016dc094, ME: 0x00000000
06-20 22:34:14.221 17094 17117 W Parcel  : Expecting binder but got null!
06-20 22:34:14.230 17094 17094 D MainActivity: onPrepareOptionsMenu() start
06-20 22:34:14.234 17094 17094 D MainActivity: onPrepareOptionsMenu() complete
06-20 22:34:14.249 17094 17094 I python.stdout: [Android] Not implemented: Canvas.new_path()

**Comment by mhsmith** [comment]
There is an Android Canvas implementation, but it has a lot of missing features. We're working on that right now, and we should have a new version available within the next few weeks.

**Reply by ccinsz** [reply]
Got it. Looking forward to it. Thank you for your answer.

**Reply by ccinsz** [reply]
Is the canvas of the android app implemented?

**Reply by freakboy3742** [reply]
Yes - the main branch now contains an implementation of Canvas on Android. It will be included in the next stable release of Toga.

**Reply by ccinsz** [reply]
Great!!!
